discovery_engine_core/core/src/engine.rs

```diff
@@ -63,11 +82,25 @@ impl<R> Engine<R>
 where
     R: Ranker,
 {
+    /// Creates a new `Engine` from configuration.
+    pub fn from_config(config: Config, ranker: R) -> Self {
+        Self {
+            config,
+            stacks: HashMap::new(), // FIXME check
+            ranker,
+        }
+    }
+
     /// Creates a new `Engine` from serialized state and stack operations.
     ///
     /// The `Engine` only keeps in its state data related to the current [`BoxedOps`].
     /// Data related to missing operations will be dropped.
-    pub fn new(state: &[u8], ranker: R, stacks_ops: Vec<BoxedOps>) -> Result<Self, Error> {
+    pub fn new(
+        state: &[u8],
+        config: Config, // FIXME part of state?
```

Comment:
no

Reply:
ok, so the `serialize` method stays exactly as it is.